File: 【哈希表】2023Q1A-删除最少字符.java
/*
 * @hw id=2023Q1A lang=java
 *
 * 删除最少字符
 */

// @hw code=start
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s = in.nextLine();
        int[] cnt = new int[26];
        Arrays.fill(cnt, Integer.MAX_VALUE);
        for(char c : s.toCharArray()) {
            int i = c - 'a';
            if(cnt[i] == Integer.MAX_VALUE) {
                cnt[i] = 0;
            }
            cnt[i] += 1;
        }
        int mn = Arrays.stream(cnt).min().getAsInt();
        StringBuilder ans = new StringBuilder();
        for(char c : s.toCharArray()) {
            int i = c - 'a';
            if(cnt[i] > mn) {
                ans.append(c);
            }
        }
        System.out.println(ans.toString());
    }
}
// @hw code=end

File: 【哈希表】2023Q1A-删除最少字符.cpp
/*
 * @hw id=2023Q1A lang=cpp
 *
 * 删除最少字符
 */

// @hw code=start
#include <iostream>
#include <vector>
#include <climits>
#include <algorithm>
using namespace std;

int main()
{
    string s;
    cin >> s;
    vector<int> cnt(26, INT_MAX);
    for (char c : s)
    {
        int i = c - 'a';
        if (cnt[i] == INT_MAX)
        {
            cnt[i] = 0;
        }
        cnt[i] += 1;
    }
    int mn = *min_element(cnt.begin(), cnt.end());
    string ans;
    for (char c : s)
    {
        int i = c - 'a';
        if (cnt[i] > mn)
        {
            ans += c;
        }
    }
    if (ans.empty())
    {
        cout << "empty" << endl;
    }
    else
    {
    cout << ans << endl;
    }
    return 0;
}
// @hw code=end